src/debugger.ts

```diff
@@ -45,14 +47,16 @@ export class Debugger extends BoxPanel {
 export namespace Debugger {
   export interface IOptions {
     connector?: IDataConnector<ReadonlyJSONValue>;
-
+    noteTracker?: INotebookTracker;
```

Comment:
I'm not sure I understand the reasoning here. Why does the `Debugger` need a reference to any trackers? The pattern I had in mind was the extension to have the trackers and as those trackers change for it to modify the debugger's handle to individual text editors.

Reply:
In actual branch, i working on a split this. I get your idea before, these changes I made after the suggestion of Jeremy. I guess tomorrow or on Wednesday I will merge changes to this branch.

Reply:
Ah okay, great; I'll take a look whenever you're ready for that!